### ISSUE TITLE
Missing package: pikepdf

### ISSUE THREAD

**Issue by jrast** [piwheels/packages#176]
Package name: pikepdf
Issue type: Missing package
Link to PyPI page: https://pypi.org/project/pikepdf
Link to piwheels page: https://www.piwheels.org/project/pikepdf
Version: > 0.9.1
Python version: 3.5+
I am the maintainer: No
More information:

pikepdf is a nice library to manipulate PDF's and is used by OCRmyPDF (which does what the name says).  It would be nice if this library is available.

**Comment by goncaloruao** [comment]
I'm also facing problems building pikepdf on raspberry pi. Is there any change to add it to piwheels.org?

Thanks ! :)

**Comment by jrast** [comment]
Installing pikepdf has the following requirements:

- a C++14 compliant compiler - GCC (5 and up), clang (3.3 and up), MSVC (2015 or newer)
- pybind11
- libqpdf 10.0.3 or higher from the QPDF project.

On Raspbian the following Versions of libqpdf are available:
- *stretch* : 6.0.0-2
- *stretch-backports*: 8.0.2-2
- *buster*: 8.4.0-2
- *bullseye*: 10.0.4-1

So, if I'm not wrong, the part which makes it difficult is that Raspbian only provides very old versions of libqpdf. Correct me if I'm wrong...

**Comment by bennuttall** [comment]
> So, if I'm not wrong, the part which makes it difficult is that Raspbian only provides very old versions of libqpdf. Correct me if I'm wrong...

Yes that seems to be the case, in which case it's probably not buildable until (perhaps) Bullseye is out.

**Comment by jrast** [comment]
If someone likes to use pikepdf, these steps can be used to install the package (tested on Raspbian Buster):
````
sudo apt install libjpeg-dev
curl -OL https://github.com/qpdf/qpdf/releases/download/release-qpdf-10.3.2/qpdf-10.3.2.tar.gz
tar -xf qpdf-10.3.2.tar.gz
cd ./qpdf-10.3.2
./configure; make; sudo make install
sudo ldconfig
pip3 install --user pikepdf  # --> This should work now!
````
This installs a new release of qpdf which is compatible with pikepdf. Note that this might break if another version of qpdf is already installed! Use at your own risk!

**Comment by MichaIng** [comment]
Latest version is not compatible with Python 3.9 anymore. Python 3.11 build fails with:
```
2026-03-18T08:51:35,657   /tmp/pip-build-env-_m5rm7dt/overlay/local/lib/python3.11/dist-packages/pybind11/include/pybind11/detail/common.h:1048:11: error: invalid use of incomplete type 'struct pybind11::detail::remove_class<auto (init_job(pybind11::module_&)::<lambda(QPDFJob&)>::*)(QPDFJob&) const>'
```
I face the very same error when trying to compile the wheel on `x86_64`. Likely on Debian Bookworm, `libqpdf` is again too old to build latest `pikepdf` version.

Trixie/Python 3.13 generally seems to succeed, but the latest build has no logs, and the one before that has a truncated log, so I cannot see the actual error.
